truffle/src/com.oracle.truffle.api/src/com/oracle/truffle/api/TruffleContext.java

```diff
@@ -1036,6 +1037,23 @@ public Builder allowPolyglotAccess(boolean b) {
             return this;
         }
 
+        /**
+         * Installs handler to control what happens on multiple thread access. When multiple threads
+         * are accessing a context which isn't ready for multithreaded access an exception is
+         * yielded by default. By installing this {@code handler} one can control what shall happen.
+         * Either to throw exception (with the provided reason) or to resolve the multithreaded
+         * situation somehow and return to retry the thread access again.
+         *
+         * @param handler callback (that gets a reason as an input) that either throws an exception
+         *            or returns to signal a <b>request for retry</b>
+         * @return this builder
+         * @since 24.2
+         */
+        public Builder threadAccessDeniedHandler(Consumer<String> handler) {
```

Comment:
Since we keep this in TruffleContext we can go back to Consumer < RuntimeException >

Reply:
I still assume that one day people demand this API in `org.graalvm.polyglot`. When that happens, it is better for the API to be a-like to this `TruffleContext` API.

I believe you were right there is no value in using `RuntimeException`. The message seems to work well.

Thus I prefer to keep `Consumer<String>` argument.

Reply:
you are right String is better.